BUG:  numpy/core/src/common/simd/vec/memory.h:113:1: error: call to 'vec_st' is ambiguous

### Describe the issue:

While compiling with Clang (13, 14, 15 and 16, I haven't tried other versions) on powerpc64le, the following error prevents successful compilation:
```
In file included from numpy/core/src/umath/loops_comparison.dispatch.c.src:12:
In file included from numpy/core/src/common/simd/simd.h:63:
In file included from numpy/core/src/common/simd/vec/vec.h:105:
numpy/core/src/common/simd/vec/memory.h:113:1: error: call to 'vec_st' is ambiguous
NPYV_IMPL_VEC_MEM(f64, f64)
^~~~~~~~~~~~~~~~~~~~~~~~~~~
numpy/core/src/common/simd/vec/memory.h:94:7: note: expanded from macro 'NPYV_IMPL_VEC_MEM'
    { npyv__storea((npyv_lanetype_##DW_CAST*)ptr, (npyv_##DW_CAST)vec); }               \
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
numpy/core/src/common/simd/vec/memory.h:42:36: note: expanded from macro 'npyv__storea'
    #define npyv__storea(PTR, VEC) vec_st(VEC, 0, PTR)
                                   ^~~~~~
/usr/lib/clang/14.0.5/include/altivec.h:11130:37: note: candidate function
static __inline__ void __ATTRS_o_ai vec_st(vector signed char __a, long __b,
                                    ^
/usr/lib/clang/14.0.5/include/altivec.h:11135:37: note: candidate function
static __inline__ void __ATTRS_o_ai vec_st(vector signed char __a, long __b,
                                    ^
/usr/lib/clang/14.0.5/include/altivec.h:11140:37: note: candidate function
static __inline__ void __ATTRS_o_ai vec_st(vector unsigned char __a, long __b,
                                    ^
/usr/lib/clang/14.0.5/include/altivec.h:11145:37: note: candidate function
static __inline__ void __ATTRS_o_ai vec_st(vector unsigned char __a, long __b,
                                    ^
/usr/lib/clang/14.0.5/include/altivec.h:11150:37: note: candidate function
static __inline__ void __ATTRS_o_ai vec_st(vector bool char __a, long __b,
                                    ^
/usr/lib/clang/14.0.5/include/altivec.h:11155:37: note: candidate function
static __inline__ void __ATTRS_o_ai vec_st(vector bool char __a, long __b,
                                    ^
/usr/lib/clang/14.0.5/include/altivec.h:11160:37: note: candidate function
static __inline__ void __ATTRS_o_ai vec_st(vector bool char __a, long __b,
                                    ^
/usr/lib/clang/14.0.5/include/altivec.h:11165:37: note: candidate function
static __inline__ void __ATTRS_o_ai vec_st(vector short __a, long __b,
                                    ^
/usr/lib/clang/14.0.5/include/altivec.h:11170:37: note: candidate function
static __inline__ void __ATTRS_o_ai vec_st(vector short __a, long __b,
                                    ^
/usr/lib/clang/14.0.5/include/altivec.h:11175:37: note: candidate function
static __inline__ void __ATTRS_o_ai vec_st(vector unsigned short __a, long __b,
                                    ^
/usr/lib/clang/14.0.5/include/altivec.h:11180:37: note: candidate function
static __inline__ void __ATTRS_o_ai vec_st(vector unsigned short __a, long __b,
                                    ^
/usr/lib/clang/14.0.5/include/altivec.h:11185:37: note: candidate function
static __inline__ void __ATTRS_o_ai vec_st(vector bool short __a, long __b,
                                    ^
/usr/lib/clang/14.0.5/include/altivec.h:11190:37: note: candidate function
static __inline__ void __ATTRS_o_ai vec_st(vector bool short __a, long __b,
                                    ^
/usr/lib/clang/14.0.5/include/altivec.h:11195:37: note: candidate function
static __inline__ void __ATTRS_o_ai vec_st(vector bool short __a, long __b,
                                    ^
/usr/lib/clang/14.0.5/include/altivec.h:11200:37: note: candidate function
static __inline__ void __ATTRS_o_ai vec_st(vector pixel __a, long __b,
                                    ^
/usr/lib/clang/14.0.5/include/altivec.h:11205:37: note: candidate function
static __inline__ void __ATTRS_o_ai vec_st(vector pixel __a, long __b,
                                    ^
/usr/lib/clang/14.0.5/include/altivec.h:11210:37: note: candidate function
static __inline__ void __ATTRS_o_ai vec_st(vector pixel __a, long __b,
                                    ^
/usr/lib/clang/14.0.5/include/altivec.h:11215:37: note: candidate function
static __inline__ void __ATTRS_o_ai vec_st(vector int __a, long __b,
                                    ^
/usr/lib/clang/14.0.5/include/altivec.h:11220:37: note: candidate function
static __inline__ void __ATTRS_o_ai vec_st(vector int __a, long __b, int *__c) {
                                    ^
/usr/lib/clang/14.0.5/include/altivec.h:11224:37: note: candidate function
static __inline__ void __ATTRS_o_ai vec_st(vector unsigned int __a, long __b,
                                    ^
/usr/lib/clang/14.0.5/include/altivec.h:11229:37: note: candidate function
static __inline__ void __ATTRS_o_ai vec_st(vector unsigned int __a, long __b,
                                    ^
/usr/lib/clang/14.0.5/include/altivec.h:11234:37: note: candidate function
static __inline__ void __ATTRS_o_ai vec_st(vector bool int __a, long __b,
                                    ^
/usr/lib/clang/14.0.5/include/altivec.h:11239:37: note: candidate function
static __inline__ void __ATTRS_o_ai vec_st(vector bool int __a, long __b,
                                    ^
/usr/lib/clang/14.0.5/include/altivec.h:11244:37: note: candidate function
static __inline__ void __ATTRS_o_ai vec_st(vector bool int __a, long __b,
                                    ^
/usr/lib/clang/14.0.5/include/altivec.h:11249:37: note: candidate function
static __inline__ void __ATTRS_o_ai vec_st(vector float __a, long __b,
                                    ^
/usr/lib/clang/14.0.5/include/altivec.h:11254:37: note: candidate function
static __inline__ void __ATTRS_o_ai vec_st(vector float __a, long __b,
```
                                    ^

### Reproduce the code example:

```python
This issue affects compilation of numpy itself.
```


### Error message:

_No response_

### Runtime information:

```
Python 3.9.16 (main, Feb 23 2023, 14:31:16)
[Clang 14.0.5 (https://github.com/llvm/llvm-project.git llvmorg-14.0.5-0-gc1238 on freebsd13
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> print(sys.version)
3.9.16 (main, Feb 23 2023, 14:31:16)
[Clang 14.0.5 (https://github.com/llvm/llvm-project.git llvmorg-14.0.5-0-gc1238
```


### Context for the issue:

_No response_

--- a/numpy/core/src/common/simd/vec/memory.h
+++ b/numpy/core/src/common/simd/vec/memory.h
@@ -85,7 +85,7 @@ NPY_FINLINE npyv_u64 npyv__loadl(const void *ptr)
     NPY_FINLINE void npyv_store_##SFX(npyv_lanetype_##SFX *ptr, npyv_##SFX vec)         \
     { npyv__store((npyv_lanetype_##DW_CAST*)ptr, (npyv_##DW_CAST)vec); }                \
     NPY_FINLINE void npyv_storea_##SFX(npyv_lanetype_##SFX *ptr, npyv_##SFX vec)        \
-    { npyv__storea((npyv_lanetype_##DW_CAST*)ptr, (npyv_##DW_CAST)vec); }               \
+    { npyv__storea((npyv_lanetype_u32*)ptr, (npyv_u32)vec); }                           \
     NPY_FINLINE void npyv_stores_##SFX(npyv_lanetype_##SFX *ptr, npyv_##SFX vec)        \
     { npyv_storea_##SFX(ptr, vec); }                                                    \
     NPY_FINLINE void npyv_storel_##SFX(npyv_lanetype_##SFX *ptr, npyv_##SFX vec)        \

--- a/numpy/distutils/ccompiler_opt.py
+++ b/numpy/distutils/ccompiler_opt.py
@@ -495,9 +495,9 @@ class _Config:
             )
             if self.cc_is_clang:
                 partial["VSX"]["flags"]  = "-maltivec -mvsx"
-                partial["VSX2"]["flags"] = "-mpower8-vector"
-                partial["VSX3"]["flags"] = "-mpower9-vector"
-                partial["VSX4"]["flags"] = "-mpower10-vector"
+                partial["VSX2"]["flags"] = "-mcpu=power8"
+                partial["VSX3"]["flags"] = "-mcpu=power9"
+                partial["VSX4"]["flags"] = "-mcpu=power10"
 
             return partial
 

--- a/numpy/distutils/tests/test_ccompiler_opt.py
+++ b/numpy/distutils/tests/test_ccompiler_opt.py
@@ -440,7 +440,7 @@ class _Test_CCompilerOpt:
             x86_iccw="/arch:SSE2",
             x86_msvc="/arch:SSE2" if self.march() == "x86" else "",
             ppc64_gcc= "-mcpu=power8",
-            ppc64_clang="-maltivec -mvsx -mpower8-vector",
+            ppc64_clang="-mcpu=power8",
             armhf_gcc="-mfpu=neon-fp16 -mfp16-format=ieee",
             aarch64="",
             s390x="-mzvector -march=arch12"

